### PR TITLE
Docker develop publishing task

### PR DESCRIPTION
**Pull request by garyschulte**
## PR description
Add a minimal github action to publish `develop` docker containers on merge to main.  

The task will publish one develop image per arch, and a multi-arch manifest.
e.g. the images that this task will publish on every merge to main (currently) are:

 
* develop multi-arch
* develop-arm64  (aarch64)
* develop-amd64 (amd64)

### Thanks for sending a pull request! Have you done the following?

- [ ] Checked out our [contribution guidelines](https://github.com/hyperledger/besu/blob/main/CONTRIBUTING.md)?
- [ ] Considered documentation and added the `doc-change-required` label to this PR [if updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).
- [ ] Considered the changelog and included an [update if required](https://wiki.hyperledger.org/display/BESU/Changelog).
- [ ] For database changes (e.g. KeyValueSegmentIdentifier) considered compatibility and performed forwards and backwards compatibility tests

### Locally, you can run these tests to catch failures early:

- [ ] unit tests: `./gradlew build`
- [ ] acceptance tests: `./gradlew acceptanceTest`
- [ ] integration tests: `./gradlew integrationTest`
- [ ] reference tests: `./gradlew ethereum:referenceTests:referenceTests`

### CODE REVIEW

**Review by usmansaleem — COMMENTED** [review]
Justin has modified the build to only use one jdk i.e. jdk 21. This gh action may be required to be updated.

**Comment by garyschulte** [comment]
> Justin has modified the build to only use one jdk i.e. jdk 21. This gh action may be required to be updated.

Thanks for the heads up.  Rebasing after the merge of @jflo 's docker changes, I removed the explicit docker variant dockerfile linting, and tested on my fork.  We now produce just 3 develop images on a merge to main:

develop (multi-arch)
develop-arm64
develop-amd64

File: .github/workflows/develop.yml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+          fi
+
+          # Get the current date and time in the format YY.MM
+          DATE_TIME=$(date +"%y.%m")
+          # Get the short SHA of the merge commit
+          SHORT_SHA=${GITHUB_SHA::7}
+          # Construct the build target name
+          BUILD_TARGET_NAME="${DATE_TIME}-develop-${SHORT_SHA}"
+          echo "Build Target Name: $BUILD_TARGET_NAME"
+          # Set the build target name as an environment variable
```
Comment (usmansaleem) [inline .github/workflows/develop.yml:61]: is there meant to be an `export` command here based on the comment?

**Review by usmansaleem — APPROVED** [review]
LGTM, minor comment.

File: .github/workflows/develop.yml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            echo "ARCH=amd64" >> $GITHUB_OUTPUT
+          else
+            echo "PLATFORM_PAIR=linux-arm64" >> $GITHUB_OUTPUT
+            echo "ARCH=arm64" >> $GITHUB_OUTPUT
+          fi
+
+          # Get the current date and time in the format YY.MM
+          DATE_TIME=$(date +"%y.%m")
+          # Get the short SHA of the merge commit
+          SHORT_SHA=${GITHUB_SHA::7}
```
Comment (siladu) [inline .github/workflows/develop.yml:57]: Is this equivalent to https://github.com/hyperledger/besu/blob/ec06ab556769103ea11c72680a09db960557b88c/.github/workflows/docker.yml#L53-L55
or is SHORT_SHA only used internally here?

If the latter, then why don't we also need the set the `sha` variable in the same way docker.yml does?

File: build.gradle [hunk 3]
```diff
@@ -979,6 +979,7 @@ def getGitCommitDetails(length = 8) {
 def isInterimBuild(dockerBuildVersion) {
   return (dockerBuildVersion ==~ /.*-SNAPSHOT/) || (dockerBuildVersion ==~ /.*-alpha/)
   || (dockerBuildVersion ==~ /.*-beta/) || (dockerBuildVersion ==~ /.*-RC.*/)
+  || (dockerBuildVersion ==~ /.*develop.*/)
```
Comment (siladu) [inline build.gradle:982]: comment above is out of date now :)

**Review by siladu — APPROVED** [review]
Couple of non-blocking comments. I took the liberty of making a one character change :)

**Pull request merged**